### PR TITLE
Rationalize Links menu items. Supports WTF-8. Internal clipboard

### PR DESCRIPTION
**Pull request by poire-z**
Followup to https://github.com/koreader/koreader/pull/4477#issuecomment-453837780:
- Remove duplicated "Show footnote popup". Have the same setting applied for Tap or 'Swipe to follow nearest link'.
- Make some menu items enabled or disabled depending on if they would have some effect with the current state of other menu items, as some kind of visual self-documentation of these dependancies.
- Add menu item to set the footnote popup font size, relative to the book font size.
- SpinWidget: allow for showing some informative text
- TouchMenu: show checkboxes in grey for disabled menu items

<kbd>image</kbd>

<kbd>image</kbd>
(wording corrections welcome :)

Also includes:
- util.splitToChars(): supports text encoded in WTF-8
WTF-8 is a superset of UTF-8, that includes UTF-16 surrogates in UTF-8 bytes (forbidden in well-formed UTF-8). https://en.wikipedia.org/wiki/UTF-8#WTF-8 https://simonsapin.github.io/wtf-8/
We may get UTF-8 with these from bad producers or converters.
We can get such chars in the text we get from Wikipedia API once their (fully valid) JSON has been decoded by our lpeg-based JSON decoder (which is a defect, hard to fix). (Our other pure-Lua json
decoder has no problem and do that correctly). More details in https://github.com/koreader/crengine/pull/252#issue-244297819
We might also find these WTF-8 in some dictionaries, so let's support them.
- Implement generic internal clipboard for all devices https://github.com/koreader/koreader/issues/4471#issuecomment-453921769
Will allow copying selected text from a book in the TextEditor.
Will be overriden by devices that implement interactions with the system clipboard.

### CODE REVIEW

**Comment by poire-z** [comment]
My quick clipboard implemenation failed luacheck:
```
accessing undefined variable '_internal_clipboard_text'
setting non-standard global variable '_internal_clipboard_text'
```
Should I make them all (in android, sdl, readerhighlight and inputtext.lua) be a method (they are functions), that would accept a `self`, so we can use `self._internal_clipboard_text` ?
Or let this commit out till someone has a better idea?

**Comment by Frenzie** [comment]
Just set it as local somewhere higher up?

**Comment by poire-z** [comment]
Oh, right, as a module local ! Thanks for reminding me the obvious :)

File: frontend/apps/reader/modules/readerlink.lua [hunk 1]
```diff
@@ -196,45 +193,77 @@ From the footnote popup, you can jump to the footnote location in the book by sw
...
+                    width = Screen:getWidth() * 0.75,
+                    value = G_reader_settings:readSetting("footnote_popup_relative_font_size") or -2,
+                    value_min = -10,
+                    value_max = 5,
+                    precision = "%+d",
+                    ok_text = _("Set font size"),
+                    title_text =  _("Set footnote popup font size"),
+                    text = _([[
+The footnote popup font adjusts to the font size you've set for the document.
+This allows you to set how smaller or bigger it should be relatively to the document font size.
```
Comment (Frenzie) [inline frontend/apps/reader/modules/readerlink.lua]:
```suggestion
This allows you to specify how much smaller or larger it should be relative to the document font size.
```

File: frontend/apps/reader/modules/readerlink.lua [hunk 2]
```diff
@@ -196,45 +193,77 @@ From the footnote popup, you can jump to the footnote location in the book by sw
...
+The recommended value is -2.]]),
+                    callback = function(spin)
+                        G_reader_settings:saveSetting("footnote_popup_relative_font_size", spin.value)
+                    end,
+                })
             end,
-            help_text = footnote_popup_help_text,
+            help_text = _([[
+The footnote popup font adjusts to the font size you've set for the document.
+This allows you to set how smaller or bigger it should be relatively to the document font size.]]),
```
Comment (Frenzie) [inline frontend/apps/reader/modules/readerlink.lua]: see above

**Pull request merged**